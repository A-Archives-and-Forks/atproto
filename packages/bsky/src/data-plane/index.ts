import { Service } from './gen/bsky_connect'
import { createPromiseClient } from '@connectrpc/connect'
import { createConnectTransport } from '@connectrpc/connect-node'

export const createDataPlaneClient = (baseUrl: string) => {
  const transport = createConnectTransport({
    baseUrl,
    httpVersion: '2',
  })
  return createPromiseClient(Service, transport)
}
