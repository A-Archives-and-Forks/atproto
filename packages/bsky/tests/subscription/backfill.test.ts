import AtpAgent from '@atproto/api'
import { SeedClient } from '../seeds/client'
import { TestNetwork } from '@atproto/dev-env'
import { forSnapshot } from '../_util'
import usersBulk from '../seeds/users-bulk'
import { chunkArray, DAY, wait } from '@atproto/common'
import { ids } from '@atproto/api/src/client/lexicons'
import { backfillRepos } from '../../src'

describe('repo subscription backfill', () => {
  let network: TestNetwork
  let agent: AtpAgent
  let pdsAgent: AtpAgent
  let sc: SeedClient
  let dids: string[]

  beforeAll(async () => {
    network = await TestNetwork.create({
      dbPostgresSchema: 'subscription_backfill',
    })
    agent = network.bsky.getClient()
    pdsAgent = network.pds.getClient()
    sc = new SeedClient(pdsAgent)
    await wait(100) // allow pending sub to be established
    await network.bsky.sub.destroy()
    await usersBulk(sc, 50)

    // For consistent ordering
    dids = Object.keys(sc.dids)
      .sort()
      .map((handle) => sc.dids[handle])

    // Place all events out of the backfill period...
    await network.pds.ctx.db.db
      .updateTable('repo_seq')
      .set({
        sequencedAt: new Date(Date.now() - 2 * DAY).toISOString(),
      })
      .execute()
    // ...except one, which we should pick-up from.
    await updateProfile(pdsAgent, dids[0], { displayName: 'ping' })
    await network.pds.ctx.sequencerLeader.isCaughtUp()
  })

  afterAll(async () => {
    await network.close()
  })

  it('ingests all repos via backfill.', async () => {
    // Ensure no profiles have been indexed
    const profilesBefore = await getAllProfiles(agent, dids)
    expect(profilesBefore).toEqual([])

    await backfillRepos(network.bsky.ctx, 9)
    // Process backfill
    network.bsky.sub.resume()
    await network.processAll(60000)

    // Check all backfilled profiles
    const profilesAfter = await getAllProfiles(agent, dids)
    expect(dids).toEqual(profilesAfter.map((p) => p.did))
    expect(forSnapshot(profilesAfter)).toMatchSnapshot()
  })

  it('continues processing after backfill.', async () => {
    await updateProfile(pdsAgent, dids[0], { displayName: 'updated' })
    await network.processAll()
    const { data: profile } = await agent.api.app.bsky.actor.getProfile({
      actor: dids[0],
    })
    expect(profile.displayName).toEqual('updated')
  })

  async function getAllProfiles(agent: AtpAgent, dids: string[]) {
    const profileChunks = await Promise.all(
      chunkArray(dids, 25).map(async (chunk) => {
        const { data } = await agent.api.app.bsky.actor.getProfiles({
          actors: chunk,
        })
        return data.profiles
      }),
    )
    return profileChunks.flat()
  }

  async function updateProfile(
    agent: AtpAgent,
    did: string,
    record: Record<string, unknown>,
  ) {
    return await agent.api.com.atproto.repo.putRecord(
      {
        repo: did,
        collection: ids.AppBskyActorProfile,
        rkey: 'self',
        record,
      },
      { headers: sc.getHeaders(did), encoding: 'application/json' },
    )
  }
})
