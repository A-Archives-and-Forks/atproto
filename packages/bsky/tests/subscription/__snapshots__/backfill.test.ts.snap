// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sync ingests all repos via backfill. 1`] = `
Array [
  Object {
    "did": "user(0)",
    "displayName": "ping",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "adrienne49.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(1)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "alden-wolff22.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(3)/cids(0)@jpeg",
    "did": "user(2)",
    "displayName": "Cody Berge",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "alec-bergnaum.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(5)/cids(0)@jpeg",
    "did": "user(4)",
    "displayName": "Winifred Harber",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "alexander-hickle.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(7)/cids(0)@jpeg",
    "did": "user(6)",
    "displayName": "Carlton Abernathy IV",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "aliya-hodkiewicz.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(9)/cids(0)@jpeg",
    "did": "user(8)",
    "displayName": "Loren Krajcik",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "aliyah-flatley12.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(11)/cids(0)@jpeg",
    "did": "user(10)",
    "displayName": "Angelina Hills",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "amy-bins.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(13)/cids(0)@jpeg",
    "did": "user(12)",
    "displayName": "Jim Kertzmann",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "angela-morissette.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(14)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "angelita-schaefer27.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(15)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "ariel-krajcik.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(17)/cids(0)@jpeg",
    "did": "user(16)",
    "displayName": "Todd Stark",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "benedict-schulist.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(18)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "bettie-bogisich96.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(19)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "bianka33.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(21)/cids(0)@jpeg",
    "did": "user(20)",
    "displayName": "Clint Ward",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "brooke24.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(22)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "brycen-smith.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(23)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "carlos6.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(25)/cids(0)@jpeg",
    "did": "user(24)",
    "displayName": "Rachel Kshlerin",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "cayla-marquardt39.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(26)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "charles-spencer.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(28)/cids(0)@jpeg",
    "did": "user(27)",
    "displayName": "Kelvin Klein",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "chauncey-klein.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(30)/cids(0)@jpeg",
    "did": "user(29)",
    "displayName": "Sammy Larkin",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "clark-stehr8.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(31)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "crystal-boyle.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(33)/cids(0)@jpeg",
    "did": "user(32)",
    "displayName": "Jeanette Weimann",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "darrion91.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(34)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "dewitt-wunsch.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(36)/cids(0)@jpeg",
    "did": "user(35)",
    "displayName": "Miss Danielle Weber",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "dock-mann91.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(38)/cids(0)@jpeg",
    "did": "user(37)",
    "displayName": "Nicole Bradtke",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "dortha-terry.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(39)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "dovie33.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(40)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "dylan61.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(42)/cids(0)@jpeg",
    "did": "user(41)",
    "displayName": "Gordon Weissnat",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "eleazar-balistreri70.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(44)/cids(0)@jpeg",
    "did": "user(43)",
    "displayName": "Alberta Fay",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "elouise28.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(46)/cids(0)@jpeg",
    "did": "user(45)",
    "displayName": "Dr. Lowell DuBuque",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "elta48.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(47)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "ervin-daugherty.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(48)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "fidel-rath.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(49)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "fredrick-mueller.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(51)/cids(0)@jpeg",
    "did": "user(50)",
    "displayName": "Myron Wolf",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "gerda-marquardt.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(53)/cids(0)@jpeg",
    "did": "user(52)",
    "displayName": "Ms. Ida Wilderman",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "gilberto-watsica65.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(54)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "gina97.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(55)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "greyson-tromp3.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(57)/cids(0)@jpeg",
    "did": "user(56)",
    "displayName": "Mrs. Angelo Keeling",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "gunnar-dare70.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(59)/cids(0)@jpeg",
    "did": "user(58)",
    "displayName": "Elsa Schaden",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "hailey-stroman.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(60)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "hank-powlowski32.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(62)/cids(0)@jpeg",
    "did": "user(61)",
    "displayName": "Rita Zieme",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "hassie-schuppe.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(64)/cids(0)@jpeg",
    "did": "user(63)",
    "displayName": "Kelli Schinner III",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "herman-gleichner95.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(65)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "hilario84.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(67)/cids(0)@jpeg",
    "did": "user(66)",
    "displayName": "Marion Dickinson",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "isai-kunze72.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(69)/cids(0)@jpeg",
    "did": "user(68)",
    "displayName": "Wilbur Schuster",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "israel-hermann0.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(70)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "jamil-batz.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(72)/cids(0)@jpeg",
    "did": "user(71)",
    "displayName": "Miss Terrell Ziemann",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "janiya48.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(73)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "javon24.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(75)/cids(0)@jpeg",
    "did": "user(74)",
    "displayName": "Patrick Sawayn",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "jeffrey-sawayn87.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(77)/cids(0)@jpeg",
    "did": "user(76)",
    "displayName": "Eileen Stroman",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "jermain-smith.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(79)/cids(0)@jpeg",
    "did": "user(78)",
    "displayName": "Ms. Darin Cole",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "joanie56.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(80)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "joel-gulgowski.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(82)/cids(0)@jpeg",
    "did": "user(81)",
    "displayName": "Erica Mertz",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "jonathan-green.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(83)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "joshuah-parker11.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(84)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "justus58.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(86)/cids(0)@jpeg",
    "did": "user(85)",
    "displayName": "Andres Towne",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "kara-zieme58.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(87)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "kaycee66.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(89)/cids(0)@jpeg",
    "did": "user(88)",
    "displayName": "Miss Floyd McClure",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "kayli-bode.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(90)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "kelton-nitzsche43.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(92)/cids(0)@jpeg",
    "did": "user(91)",
    "displayName": "Sergio Hane",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "kendra-ledner80.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(93)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "kiana-schmitt39.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(94)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "kiera97.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(95)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "kim41.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(96)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "kyle72.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(97)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "leann49.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(98)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "lennie-strosin.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(100)/cids(0)@jpeg",
    "did": "user(99)",
    "displayName": "Karl Bosco IV",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "leonel-koch43.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(101)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "lilliana-waters.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(102)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "lura-jacobi55.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(104)/cids(0)@jpeg",
    "did": "user(103)",
    "displayName": "Sally Funk",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "magnus53.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(105)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "maiya42.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(106)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "marietta-zboncak.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(107)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "moises-bins8.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(109)/cids(0)@jpeg",
    "did": "user(108)",
    "displayName": "Mrs. Clifford Mertz",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "murphy35.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(110)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "nannie18.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(112)/cids(0)@jpeg",
    "did": "user(111)",
    "displayName": "Dean Romaguera",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "natasha83.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(114)/cids(0)@jpeg",
    "did": "user(113)",
    "displayName": "Travis Hoppe",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "nestor90.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(115)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "nicolas-krajcik10.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(116)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "ola-oconnell.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(117)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "paris-swift.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(118)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "polly-shanahan45.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(119)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "precious-fay.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(121)/cids(0)@jpeg",
    "did": "user(120)",
    "displayName": "Ms. Bradford Thiel",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "preston-harris.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(122)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "raleigh-metz.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(124)/cids(0)@jpeg",
    "did": "user(123)",
    "displayName": "Pablo Schmidt",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "reina-runte33.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(125)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "reynold-ortiz.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(126)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "rhianna-stamm29.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(127)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "roderick-dibbert.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(129)/cids(0)@jpeg",
    "did": "user(128)",
    "displayName": "Yolanda VonRueden",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "rodger-maggio24.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(131)/cids(0)@jpeg",
    "did": "user(130)",
    "displayName": "Karl Goyette",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "rosamond38.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(133)/cids(0)@jpeg",
    "did": "user(132)",
    "displayName": "Alexander Purdy",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "ryley-kerluke.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(135)/cids(0)@jpeg",
    "did": "user(134)",
    "displayName": "Melissa Johnson",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "santa-hermann78.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(137)/cids(0)@jpeg",
    "did": "user(136)",
    "displayName": "Darnell Kuvalis",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "savannah-botsford.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(139)/cids(0)@jpeg",
    "did": "user(138)",
    "displayName": "Tanya Denesik",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "silas77.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(140)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "sven70.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(141)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "sydni48.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(142)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "tiara-mohr.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "avatar": "https://bsky.public.url/image/sig()/rs:fill:1000:1000:1:0/plain/user(144)/cids(0)@jpeg",
    "did": "user(143)",
    "displayName": "Alex Johnson",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "tobin63.test",
    "indexedAt": "1970-01-01T00:00:00.000Z",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(145)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "tracy56.test",
    "labels": Array [],
    "postsCount": 0,
  },
  Object {
    "did": "user(146)",
    "followersCount": 0,
    "followsCount": 0,
    "handle": "tressa-senger.test",
    "labels": Array [],
    "postsCount": 0,
  },
]
`;
